_posts/2018-08-09-deploying-esxi-through-foreman.md

```diff
@@ -341,17 +345,20 @@ Now we have to create an entry for this bootloader in the Foreman UI just like w
   chmod u+x 01-prep-esxi-uefimode.sh
 
   # Make it accessible to Foreman
-  cd /usr/share/foreman/config/hooks
-  chown -R foreman:foreman host/
+  cd /usr/share/foreman/config/
+  chown -R foreman:foreman hooks/
 
   # Register the hook script in Foreman by restarting Apache
   service apache2 restart
+
+  # Make /var/lib/tftpboot writable to foreman user as hook scripts are run using that account
+  chown foreman /var/lib/tftpboot
   ```
 
   **Important Note:** after_build hook scripts are not run on newly created hosts. These are only run when build mode is enabled on an existing host.
 
-* **3.4:** Now we are ready to deploy the host. Edit the host entry in Foreman and set the following:
-
+* **3.4:** Now we are ready to deploy the host. Edit the host entry in Foreman and set the following:  
+  Puppet Master and Puppet CA set to Foreman Server  
```

Comment:
Since we are using **host.puppet_proxy_name** in 01-prep-esxi-uefimode.sh hook script to get the FQDN on Foreman, we need to set these in the host. Otherwise the value will be null. Ofcourse this means we need to install the Puppetmaster role in Foreman too.